test/unit_tests/braket/circuits/test_free_parameter_expression.py

```diff
@@ -0,0 +1,123 @@
+# Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"). You
+# may not use this file except in compliance with the License. A copy of
+# the License is located at
+#
+#     http://aws.amazon.com/apache2.0/
+#
+# or in the "license" file accompanying this file. This file is
+# distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF
+# ANY KIND, either express or implied. See the License for the specific
+# language governing permissions and limitations under the License.
+
+import pytest
+
+from braket.circuits import FreeParameter, FreeParameterExpression
+
+
+@pytest.fixture
+def free_parameter_expression():
+    return FreeParameterExpression(FreeParameter("theta") - 1)
+
+
+def test_is_free_param_expr(free_parameter_expression):
+    assert isinstance(free_parameter_expression, FreeParameterExpression)
+
+
+def test_equality():
+    expr_1 = FreeParameterExpression(FreeParameter("theta") + 1)
+    expr_2 = FreeParameterExpression(FreeParameter("theta") + 1)
+    other_expr = FreeParameterExpression(FreeParameter("theta"))
+    non_expr = "non circuit"
+
+    assert expr_1 == expr_2
+    assert expr_1 is not expr_2
+    assert expr_1 != other_expr
+    assert expr_1 != non_expr
+
+
+def test_commutativity():
+    add_1 = 1 + FreeParameterExpression(FreeParameter("theta"))
+    add_2 = FreeParameterExpression(FreeParameter("theta")) + 1
+    mul_1 = FreeParameterExpression(FreeParameter("theta") * 1)
+    mul_2 = FreeParameterExpression(1 * FreeParameter("theta"))
+
+    assert add_1 == add_2
+    assert mul_1 == mul_2
+
+
+def test_add():
+    add_expr = FreeParameter("theta") + FreeParameter("alpha")
+    expected = FreeParameterExpression(FreeParameter("theta")) + FreeParameterExpression(
+        FreeParameter("alpha")
+    )
+
+    assert add_expr == expected
```

Comment:
The assertions here bug me a little bit, because you're verifying that the two calls are identical (using the add operator or adding two free parameter expressions) but you're not actually testing that the "add" operation is correct.